### PR TITLE
hack for hosting on ipfs

### PR DESCRIPTION
**Pull request by Velenir**
This isn't ideal and modifies base.href before scripts are injected
A better solution would be to adopt HashRouter, which also is not ideal

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr351--dexreact.review.gnosisdev.com

File: src/html/index.html [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+  <%if (htmlWebpackPlugin.options.ipfsHack) { %>
+  <script>
+    (function () {
+      const { origin, pathname } = window.location
+      const ipfsMatch = /.*\/Qm\w{44}/.exec(pathname)
+      const base = document.createElement('base')
+      let baseHref
+      if (ipfsMatch) {
+
+        baseHref = origin + ipfsMatch[0]
```
Comment (anxolin) [inline src/html/index.html]: Why is the origin required, cannot be relative for example to the domain root?

For example:
* **For url**: https://gateway.ipfs.io/ipfs/QmeYYwD4y4DgVVdAzhT7wW5vrvmbKPQj8wcV2pAzjbj886/docs/getting-started/
* **Base href would be**:`/ipfs/QmeYYwD4y4DgVVdAzhT7wW5vrvmbKPQj8wcV2pAzjbj886/`

File: src/html/index.html [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+        if (!baseHref.endsWith('/')) baseHref += '/'
+
+        window.BASE_URL = ipfsMatch[0]
```
Comment (anxolin) [inline src/html/index.html]: Do we really need the global? can we use the `<base /> href` instead?

Comment (Velenir) [reply src/html/index.html]: `base.href` is same as `document.baseURI`, it prepends origin. We'd have to recalculate anyway

**Comment by Velenir** [comment]
> * Don't do the hack using a helper/util that is used in `App.tsx` to initialize the app, so there's no need to modify and parametrize the template

Can't. `<scripts src=*>` in html are unaffected by runtime, need a hack applied before they are parsed by the browser

File: src/html/index.html (hunk 1 above)
Comment (Velenir) [reply src/html/index.html]: Not required, no real reason I included it.

**Comment by Velenir** [comment]
This is still not an ideal solution as you can't open a page relative to root. Because ipfs doesn't have 404 redirects or any other way to implement historyApi

i.e. https://gateway.ipfs.io/ipfs/Qm***/trade won't work

**Comment by anxolin** [comment]
@Velenir what about, instead of this hack we decide to use a different router if we see that we are in a IPFS server?

**Pull request merged**